crates/sim2h/src/lib.rs

```diff
@@ -62,35 +95,114 @@ pub struct Sim2h {
 }
 
 impl Sim2h {
-    pub fn new(
-        crypto: Box<dyn CryptoSystem>,
-        stream_manager: StreamManager<std::net::TcpStream>,
-        bind_spec: Lib3hUri,
-    ) -> Self {
+    pub fn new(crypto: Box<dyn CryptoSystem>, bind_spec: Lib3hUri) -> Self {
+        let pool = Pool::new();
+        pool.push_job(Box::new(Arc::new(Mutex::new(Tick::new()))));
+
+        let (wss_send, wss_recv) = crossbeam_channel::unbounded();
+        let (msg_send, msg_recv) = crossbeam_channel::unbounded();
+
         let mut sim2h = Sim2h {
             crypto,
             bound_uri: None,
             connection_states: RwLock::new(HashMap::new()),
             spaces: HashMap::new(),
-            stream_manager,
+            pool,
+            wss_recv,
+            msg_send,
+            msg_recv,
+            open_connections: HashMap::new(),
             num_ticks: 0,
             rrdht_arc_radius_recalc: std::time::Instant::now(),
             missing_aspects_resync: std::time::Instant::now(),
         };
 
-        debug!("Trying to bind to {}...", bind_spec);
-        let url: url::Url = bind_spec.clone().into();
-        sim2h.bound_uri = Some(
-            sim2h
-                .stream_manager
-                .bind(&url)
-                .unwrap_or_else(|e| panic!("Error binding to url {}: {:?}", bind_spec, e))
-                .into(),
-        );
+        sim2h.priv_bind_listening_socket(url::Url::from(bind_spec).into(), wss_send);
```

Comment:
Use url2 directly?

Reply:
sim2h was built with Lib3hUri - there's no direct `From` for url2